### DISCUSSION TITLE
Compiling for Raspberry Pi (linux-arm64)

### DISCUSSION THREAD

**Discussion by Yveske1966** [AvaloniaUI/Avalonia#16678] [Q&A] [answered]
[demo 1&2.zip](https://github.com/user-attachments/files/16596615/demo.1.2.zip)
I've made 2 demo programs by clicking create new project (see attachment)
- Demo with preset Avalonia.NET MVVM App (AvaloniaUI)
- Demo2 with preset Avalonia Cross Platform Application (AvaloniaUI)

I've changed the net version to 8.0 in all csproj with a text editor.
All NuGet packages were updated to the latest version.

I've compiled both via the command line with "dotnet publish -c release -r linux-arm64 --self-contained".
I've used the Desktop branch for the Cross Platform Demo2.

I've copied both Publish directories onto a Raspberry Pi 4.
I've tried to start the programs using "dotnet demo.dll" (or demo2.dll)
- Single platform version (Demo) runs and shows "Welcome to Avalonia"
- Cross platform version (Demo2) fails with "could not find entry point"

How can I make the Cross platform version to work ?

Visual Studio 22 version 17.11.0 Preview 7.0
Avalonia 11.1.3
Raspberry Pi 4 with Linux 12 (Bookworm) 64bit

Edit: Removed unwanted indent after first bullet list

**Comment by thevortexcloud** [comment]
Which project are you trying to run for the multi project app? Linux does not have a specific TFM in .NET. So you have to use the generic one.

**Reply by Yveske1966** [reply]
I've run the dotnet publish command from within the .Desktop directory.

**Reply by thevortexcloud** [reply, marked as answer]
That should work. My next guess is you are running the wrong dll. The executable one will be something like `demo2.Desktop.dll`. I think the one you are trying to run is the shared library which can't be directly executed.

**Reply by Yveske1966** [reply]
A very very big thank you !

**Reply by thevortexcloud** [reply]
You are welcome. Also FYI, you don't have to launch the DLL via the `dotnet` command if you are doing a proper compile. You can just launch the app binary which will deal with it for you. Binaries on Linux typically don't have an extension. So it's probably called something like `demo2.Desktop`. Which itself is a problem because a lot of desktops will see it as a `.desktop` file, not a binary file. So double clicking it will open a text editor instead of launching the app. You are probably better off just renaming the project/build output in this situation.

**Reply by Yveske1966** [reply]
Thanks again ! Works like a charm !!!
